### PR TITLE
ref(TS) Convert base/tracks to TS

### PR DESCRIPTION
**Pull request by robertpin**

### CODE REVIEW

File: react/features/base/tracks/actions.ts [hunk 1]
```diff
@@ -434,13 +437,13 @@ export function trackAdded(track) {
...
-                    dispatch(setNoSrcDataNotificationUid(notificationAction.uid));
+                    dispatch(setNoSrcDataNotificationUid(notificationAction?.uid));
 
-                    noDataFromSourceNotificationInfo = { uid: notificationAction.uid };
+                    noDataFromSourceNotificationInfo = { uid: notificationAction?.uid };
                 } else {
                     const timeout = setTimeout(() => dispatch(
                         showNoDataFromSourceVideoError(track)),
-                        NOTIFICATION_TIMEOUT_TYPE.MEDIUM);
+                        NOTIFICATION_TIMEOUT.MEDIUM);
```
Comment (saghul) [inline react/features/base/tracks/actions.ts:446]: nice catch!

File: react/features/base/tracks/functions.ts [hunk 2]
```diff
@@ -147,7 +152,10 @@ export function createLocalTracksF(options = {}, store) {
     if (typeof APP !== 'undefined') {
         // TODO The app's settings should go in the redux store and then the
         // reliance on the global variable APP will go away.
-        store || (store = APP.store); // eslint-disable-line no-param-reassign
+        // store || (store = APP.store); // eslint-disable-line no-param-reassign
```
Comment (saghul) [inline react/features/base/tracks/functions.ts:155]: drop this commented line

File: react/features/base/tracks/functions.ts [hunk 3]
```diff
@@ -159,6 +167,9 @@ export function createLocalTracksF(options = {}, store) {
         }
     }
 
+    if (!store) {
```
Comment (saghul) [inline react/features/base/tracks/functions.ts:170]: I don't think this valid in RN, is it?

**Review by saghul — COMMENTED** [review]
A couple of minor comments

File: react/features/base/tracks/functions.ts (hunk 3 above)
Comment (robertpin) [reply react/features/base/tracks/functions.ts]: I'll remove it

**Review by saghul — APPROVED** [review]

**Pull request merged**